contract/models/account_analytic_account.py

```diff
@@ -267,7 +268,7 @@ def recurring_create_invoice(self):
             ref_date = contract.recurring_next_date or fields.Date.today()
             if (contract.date_start > ref_date or
                     contract.date_end and contract.date_end < ref_date):
-                if self.env.context.get('cron'):
+                if self.env.context.get('cron'):  # pragma: no cover
```

Comment:
IMO this is cheating & should be removed. We should either take the test hit if we don't want to test it, or add the test (I'm good with either)

Reply:
No, this is not cheating in this case. It's a case that doesn't modify anything and I don't want to be reported as a coverage failure.

Reply:
There is still logic here, that if untested should yield a lack of coverage. Whether it modifies anything or not, the logic is still in the program and still does something different from the other things. 

I'm not arguing against the lack of testing for this, as I agree it's a minimal point and probably doesn't need it. I am arguing against the false coverage report though, because this is not covered and is a unique logic path that should be tested per TDD standards.

Reply:
Dave, not for this case. There's no possibility of leading to that situation right now with current constraints, so it's only a safeguard for already installed DBs, but no sense of testing that specific case (forcing through SQL) or to report as not covered.

Reply:
Untouchable code is still untested IMO. What's the harm in having it show what it is? Hiding the fact that the block of code wasn't executed during testing is just sweeping the fact it can't be touched under the rug. There's always going to be scenarios like this, which is why 100% test coverage is an unrealistic goal in the practical sense.

I won't block for it - but I really feel that `pragma: no cover` should align explicitly with the example case provided in Coverage.py docs, which is to exclude debug code. Any production code that's not actually tested should show as untested.

Reply:
OK, I give up due to your insistence...